js/edge.js

```diff
@@ -0,0 +1,812 @@
+   * @return {!ServiceBuilder} A self reference.
+   */
+  setAdbPort(port) {
+    return this.addArguments('--adb-port=' + port);
+  }
+
+  /**
+   * Sets the path of the log file the driver should log to. If a log file is
+   * not specified, the driver will log to stderr.
+   * @param {string} path Path of the log file to use.
+   * @return {!ServiceBuilder} A self reference.
+   */
+  loggingTo(path) {
+    return this.addArguments('--log-path=' + path);
+  }
+
+  /**
+   * Enables verbose logging.
+   * @return {!ServiceBuilder} A self reference.
+   */
+  enableVerboseLogging() {
+    return this.addArguments('--verbose');
+  }
+
+  /**
+   * Sets the number of threads the driver should use to manage HTTP requests.
+   * By default, the driver will use 4 threads.
+   * @param {number} n The number of threads to use.
+   * @return {!ServiceBuilder} A self reference.
+   */
+  setNumHttpThreads(n) {
+    return this.addArguments('--http-threads=' + n);
+  }
+
+  /**
+   * @override
+   */
+  setPath(path) {
+    super.setPath(path);
+    return this.addArguments('--url-base=' + path);
+  }
+}
+
+
+/** @type {remote.DriverService} */
+let defaultService = null;
+
+
+/**
+ * Sets the default service to use for new MicrosoftEdgeDriver instances.
+ * @param {!remote.DriverService} service The service to use.
+ * @throws {Error} If the default service is currently running.
+ */
+function setDefaultService(service) {
+  if (defaultService && defaultService.isRunning()) {
+    throw Error(
+      'The previously configured EdgeDriver service is still running. ' +
+      'You must shut it down before you may adjust its configuration.');
+  }
+  defaultService = service;
+}
+
+
+/**
+ * Returns the default MicrosoftEdgeDriver service. If such a service has
+ * not been configured, one will be constructed using the default configuration
+ * for an MicrosoftEdgeDriver executable found on the system PATH.
+ * @return {!remote.DriverService} The default MicrosoftEdgeDriver service.
+ */
+function getDefaultService() {
+  defaultService = new ServiceBuilder().build();
+  return defaultService;
+}
+
+
+/**
+ * Creates a new WebDriver client for Microsoft's Edge.
+ */
+class Driver extends webdriver.WebDriver {
+  /**
+   * Creates a new browser session for Microsoft's Edge browser.
+   *
+   * @param {(capabilities.Capabilities|Options)=} opt_config The configuration
+   *     options.
+   * @param {remote.DriverService=} opt_service The session to use; will use
+   *     the {@linkplain #getDefaultService default service} by default.
+   * @param {promise.ControlFlow=} opt_flow The control flow to use, or
+   *     {@code null} to use the currently active flow.
+   * @return {!Driver} A new driver instance.
+   */
+  static createSession(opt_config, opt_service, opt_flow) {
+    let executor, service, client, caps;
+    if (opt_config instanceof Options && opt_config.getEdgeChromium()) {// chromium edge
```

Comment:
Users may pass in a raw Object or a Capabilities instead of an Options and we need to handle all cases. Please take a look at the [useEdgeChromium](https://github.com/SeleniumHQ/selenium/blob/502fccb442a7496fb6bd028f35b814a80c80edbf/javascript/node/selenium-webdriver/edge.js#L144) function in Selenium 4 and use the same checks here.

Reply:
You'll also need to add a case for ``o instanceof Options`` since Options doesn't extend Capabilities in Selenium 3.

Reply:
updated to consider Options|Capabitilities|Object scenarios in the check.